dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/validator/relationship/RelationshipValidatorTest.java

```diff
@@ -0,0 +1,191 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.validation.validator.relationship;
+
+import static org.hisp.dhis.tracker.validation.ValidationCode.E1048;
+import static org.hisp.dhis.tracker.validation.ValidationCode.E1124;
+import static org.hisp.dhis.tracker.validation.ValidationCode.E4000;
+import static org.hisp.dhis.tracker.validation.ValidationCode.E4001;
+import static org.hisp.dhis.tracker.validation.ValidationCode.E4012;
+import static org.hisp.dhis.tracker.validation.validator.AssertValidations.assertHasError;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+
+import org.hisp.dhis.common.CodeGenerator;
+import org.hisp.dhis.relationship.RelationshipConstraint;
+import org.hisp.dhis.relationship.RelationshipEntity;
+import org.hisp.dhis.relationship.RelationshipType;
+import org.hisp.dhis.tracker.TrackerIdSchemeParams;
+import org.hisp.dhis.tracker.TrackerImportStrategy;
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.MetadataIdentifier;
+import org.hisp.dhis.tracker.domain.Relationship;
+import org.hisp.dhis.tracker.domain.RelationshipItem;
+import org.hisp.dhis.tracker.preheat.TrackerPreheat;
+import org.hisp.dhis.tracker.validation.Reporter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith( MockitoExtension.class )
+class RelationshipValidatorTest
+{
+
+    private RelationshipValidator validator;
+
+    @Mock
+    private TrackerPreheat preheat;
+
+    @Mock
+    private TrackerBundle bundle;
+
+    private Reporter reporter;
+
+    @BeforeEach
+    public void setUp()
+    {
+        validator = new RelationshipValidator();
+
+        TrackerIdSchemeParams idSchemes = TrackerIdSchemeParams.builder().build();
+        reporter = new Reporter( idSchemes );
+
+        when( bundle.getStrategy( any( Relationship.class ) ) ).thenReturn( TrackerImportStrategy.CREATE );
+    }
+
+    @Test
+    void shouldFailWhenRelationshipUidDoesNotHaveCorrectFormat()
+    {
+        Relationship relationship = new Relationship();
+        relationship.setRelationship( "wrong uid" );
+        when( bundle.getRelationships() ).thenReturn( List.of( relationship ) );
+
+        validator.validate( reporter, bundle, bundle );
+
+        assertEquals( 1, reporter.getErrors().size(),
+            "Expected just one error when uid not correct, but " + reporter.getErrors().size() + " found" );
```

Comment:
We will get the actual size printed by `assertEquals` what we do not get is what errors we actually got. Printing `reporter.getErrors()` would be helpful 😄